### PR TITLE
task: auth sockets

### PR DESCRIPTION
**Pull request by KubaBaniak**

### CODE REVIEW

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -23,6 +23,10 @@ export class AuthService {
     private readonly jwtService: JwtService,
   ) {}
 
+  async verifyJwt(jwtToken: string): Promise<any> {
```
Comment (PrzerostNadSens) [inline src/auth/auth.service.ts:26]: change this type to `unknown` or to a specific type if you can determine it, try not to use it `any`

Comment (PrzerostNadSens) [reply src/auth/auth.service.ts:26]: you can add interface `AccessTokenPayload`

File: src/websocket/notification.gateway.ts [hunk 2]
```diff
@@ -1,35 +1,60 @@
...
-} from '@nestjs/websockets';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { WebSocketGateway, WebSocketServer } from '@nestjs/websockets';
 import { Server, Socket } from 'socket.io';
+import { AuthService } from '../auth/auth.service';
+import { UserRepository } from '../user/user.repository';
 
 @Injectable()
 @WebSocketGateway()
 export class NotificationGateway {
```
Comment (PrzerostNadSens) [inline src/websocket/notification.gateway.ts:9]: in my opinion, it is better to change the name of it to `WebSocketGateway` (and rename the entire file to `WebSocketGateway`)

```suggestion
export class WebSocketGateway {
```

Comment (PrzerostNadSens) [reply src/websocket/notification.gateway.ts:9]: you should not connect WebSocket with only notifications 

it is better call it as  WebSocket events

File: src/websocket/notification.gateway.ts [hunk 3]
```diff
@@ -1,35 +1,60 @@
...
 
-  getNotificationRoomName(): string {
-    return `recipe-notification-room`;
+  getRoomWithId(userId: number): string {
+    return `notification-userId-${userId}`;
+  }
+
+  async handleConnection(socket: Socket) {
+    const token: string =
+      socket.handshake.headers['authorization'].split(' ')[1];
```
Comment (PrzerostNadSens) [inline src/websocket/notification.gateway.ts:24]:
```suggestion
      socket.handshake.headers.authorization;
```
it should also work, you can try it

File: src/websocket/notification.gateway.ts [hunk 4]
```diff
@@ -1,35 +1,60 @@
...
-  @SubscribeMessage('room')
-  joinRoom(@ConnectedSocket() client: Socket): void {
-    client.join(this.getNotificationRoomName());
+  addUserToNotificationRoom(socket: Socket, userId: number): void {
+    socket.join(this.getRoomWithId(userId));
   }
 
-  sendRecipeNotification(message: string): boolean {
-    return this.server.emit('notification', message);
+  sendRecipeNotification(message: string, authorId: number): boolean {
```
Comment (PrzerostNadSens) [inline src/websocket/notification.gateway.ts:48]:
```suggestion
  sendRecipeWebsocketEvent(message: string, authorId: number): boolean {
```

File: src/websocket/notification.gateway.ts [hunk 5]
```diff
@@ -1,35 +1,60 @@
...
+
+  private dissconnectOnAuth(socket: Socket) {
+    socket.emit('Error', new UnauthorizedException());
+    socket.disconnect();
   }
 
-  @SubscribeMessage('room')
-  joinRoom(@ConnectedSocket() client: Socket): void {
-    client.join(this.getNotificationRoomName());
+  addUserToNotificationRoom(socket: Socket, userId: number): void {
```
Comment (PrzerostNadSens) [inline src/websocket/notification.gateway.ts:44]:
```suggestion
  addUserToWebsocketEventRoom(socket: Socket, userId: number): void {
```

**Review by PrzerostNadSens — APPROVED** [review]

File: src/websocket/notification.gateway.ts (hunk 2 above)
Comment (KubaBaniak) [reply src/websocket/notification.gateway.ts:9]: Maybe something like `WebSocketEventGateway()` is good enough?

Comment (PrzerostNadSens) [reply src/websocket/notification.gateway.ts:9]: I think`WebSocketEventGateway()` will fit

**Pull request merged**